asterisk/agi/library/Agi/Action/ExternalFriendCallAction.php

```diff
@@ -70,8 +70,27 @@ public function process()
             return;
         }
 
-        // Outgoing presentation
-        $ddi = $friend->getOutgoingDDI();
+        // Allow identification from any company DDI
+        $callerIdNum = $this->agi->getCallerIdNum();
+        $companyDDIs = $friend->getCompany()->getDDIs();
+        foreach ($companyDDIs as $companyDDI) {
+            // FIXME This does not support E.164 DDIs
```

Comment:
I think that PAI/RPID number format should follow Friend's Country format. That way, using preferredToE164 function the result can be matched against E.164 DDI.

e.g. Friend country is Spain ---> PAI/RPID should follow Spain's calling format to be detected as a valid source.